src/client.ts

```diff
@@ -789,36 +796,72 @@ export class McpdClient {
   async getAgentTools(options?: {
     format?: "array";
     servers?: string[];
+    tools?: string[];
   }): Promise<AgentFunction[]>;
   async getAgentTools(options: {
     format: "object";
     servers?: string[];
+    tools?: string[];
   }): Promise<Record<string, AgentFunction>>;
   async getAgentTools(options: {
     format: "map";
     servers?: string[];
+    tools?: string[];
```

Comment:
nit: instead of making `tools` optional with `?` which makes it type `string[] | undefined` why not just give it a default value of `[]` instead maintaining a single type?